design-proposals/vpro-eim.md

```diff
@@ -0,0 +1,166 @@
+
+We introduce a new resource to abstract Baseboard Management Controller `BMCResource` and "connected" to the
+[Host](https://github.com/open-edge-platform/infra-core/blob/main/inventory/api/compute/v1/compute.proto#L47) with the
+cardinality 0...M.
+
+- `BaremetalControllerKind` - identifies the type of Device Management
+- `BmcIP` - the IP of the Device Management controller
+- `BmcUsername` - the username to access the Device Management controller
+- `BmcPassword` - the password to access the Device Management controller
+
+Inventory should not allow the removal of an host if one of the baremetal controller is not nil.
+
+[Host](https://github.com/open-edge-platform/infra-core/blob/main/inventory/api/compute/v1/compute.proto#L47) will be
+extended as follow:
+
+- `BootIP` - IP address learned during the device discovery
+- `BootMac` - Mac address learned during the device discovery
+- `DesiredPowerState` ->> {UNSPECIFIED, ON, SLEEP, OFF, HIBERNATE, RESET}
+- `CurrentPowerState` ->> {UNSPECIFIED, ON, SLEEP, OFF, HIBERNATE, RESET}
+- `PowerStatus` ->> is a modern status with some well known messages {Powered On, Light Sleeping, Deep Sleeping,
+Powered Off, Hibernated}
+- `PowerCommandPolicy` ->> {IMMEDIATE_OFF, ORDERED_OFF}
+- `PowerOnTime` ->> the time of last startup (from which `Uptime` could be calculated by the UI)
+- `AMTSKU` ->> a string reporting AMT/ISM version. `Unsupported` otherwise.
+- `AMTDesiredState` ->> {UNSPECIFIED, PROVISION, UNPROVISION, DISCONNECT}
+- `AMTCurrentState` ->> {UNSPECIFIED, PROVISION, UNPROVISION, DISCONNECT}
+- `AMTStatus` ->> {PROVISIONED, UNPROVISIONED, DISCONNECTED, UNSUPPORTED}
+
+**Note:** `AMTSKU` is populated during the device discovery. AMT version decoding is quite complex see
+[here][rpc-decoding]. For this reason a string is suggested instead of an Enum.
+
+AMT does not provide any info about in progress operations, nor if a reboot is in-progress. However, The DM Resource Manager can fake the `IN_PROGRESS` statuses until the state in the DMT stack does not reflect the required operation.
+For example during the provisioning, the RM can keep the `IN_PROGRESS` state until the device is connected, move to an error status if the devices does not show up after a given threshold or can fake the `IN_PROGRESS` statuses using `Last*` timestamps exposed by MPS.
+
+The behavior of the resource manager is driven by the `DesiredPowerState` and `PowerCommandPolicy`, and internally keeps a
+timer between soft and hard when `ORDERED_OFF` is set in `PowerCommandPolicy`. The PowerStatus would be updated with this internal behavior in the DM RM, for example: "Soft power off succeeded", "Soft power off failed after 30s, forcing Hard power off", etc.
+
+Instead, UI or Users can fetch directly from MPS REST APIs (through the MT-GW), using device UUID, the following
+additional information such as:
+
+- `SoftwareVersion` - AMT version deployed in the device
```

Comment:
Should we call it AMT or OOB if we are supporting ISM as well ?

Reply:
this is fetched from MPS API we dont own the schema